### ISSUE TITLE
`getItems` does not work with `prefixStorage`

### ISSUE THREAD

**Issue by harlan-zw** [unjs/unstorage#396]
### Environment

1.10.1

### Reproduction

https://stackblitz.com/edit/stackblitz-starters-rrstqp?file=index.js

### Describe the bug

Working

```ts
useStorage().getItems(['namespace:key'])
```

Not Working

```ts
prefixStorage(useStorage(), 'namespace').getItems(['key'])
useStorage('namespace').getItems(['key'])

// { key: "key", value: null }
```

It seems to switch back to the default driver (`memory`). I'm guessing this is because the mount is being discovered from the input key, without the base being applied first.

**Comment by pi0** [comment]
Can you please make a sandbox ideally with unstorage only? 🙏🏼

**Comment by harlan-zw** [comment]
Sorry I thought it was self-evident but it seems like something weird is going on.

Check this repro: https://stackblitz.com/edit/stackblitz-starters-rrstqp?file=index.js

The prefixStorage is not even working here :thinking: This works in Nitro though

**Comment by harlan-zw** [comment]
Yes, sorry thought it was clear enough. Here is a repro: https://stackblitz.com/edit/stackblitz-starters-rrstqp?file=index.js

Would be great to add a starter to readme for quickly spinning one up.

**Comment by dimasxp** [comment]
useStorage('namespace').clear() also not working with namespace

**Comment by Mahdhir** [comment]
Same issue here.
prefix storage clear doesn't work
Also clear with base doesn't work for in-memory storage
